[Console] Remove use of jQuery

mibragimov opened the pull request:
Closes https://github.com/elastic/kibana/issues/75861

### Summary

This PR addresses removing jQuery from Console.

alexwizp commented:
@mibragimov please add more detailed description for your PR's. It is important for me to see the part that has been changed in order to make it easier to review the code.

alexwizp commented on src/plugins/console/public/application/hooks/use_send_current_request_to_es/send_request_to_es.ts:
```diff
@@ -6,6 +6,8 @@
  */
 
+import { HttpSetup } from 'kibana/public';
+import _ from 'lodash';
```
please don't do imports in that way. 
Legal way: `import { isUndefined } from 'lodash';`
But honestly not sure that we need lodash here at all

alexwizp commented:
@elasticmachine merge upstream

alexwizp commented on src/plugins/console/public/lib/mappings/mappings.js:
```diff
@@ -330,74 +341,64 @@ export function retrieveAutoCompleteInfo(settings, settingsToRetrieve) {
-          mappingsResponse = '[{}]';
-        } else {
-          mappingsResponse = mappings[0];
+  const promises = Object.keys(settingKeyToPathMap).map((settingKey) => {
```
I see you tried to refactor that code. My recommendation here make that code more maintainable and use the previous approach with some modification.

I would like to see little unified methods like `retrieveDataStreams` or `retrieveAliases`

Something like: 

```
  const retrieveDataStreams = async (settingsToRetrieve) => {
    const dataStreams = await retrieveSettings('dataStreams', settingsToRetrieve);

    if (dataStreams) {
      loadDataStreams(dataStreams);
    }
  };

  Promise.allSettled([retrieveDataStreams()]).then(() => {
    // Schedule next request.
    pollTimeoutId = setTimeout(() => {
      // This looks strange/inefficient, but it ensures correct behavior because we don't want to send
      // a scheduled request if the user turns off polling.
      if (settings.getPolling()) {
        retrieveAutoCompleteInfo(http, settings, settings.getAutocomplete());
      }
    }, settings.getPollInterval());
  })
```

And please use more TS features to make our code a little bit more stable

alexwizp commented on src/plugins/console/public/lib/mappings/mappings.js:
```diff
@@ -218,34 +228,33 @@ export function loadDataStreams(data) {
   });
 }
 
 export function loadAliases(aliases) {
```
related to my previous comment. Maybe it make sense to create a separate files for each entries an use one interface for loading / initialization of each entry?

My motivation here to refactor that code to TS and make it a little bit more readable 

@cjcenizal @sabarasaba

mibragimov replied:
Created a separate issue for converting the file to ts https://github.com/elastic/kibana/issues/129716

alexwizp commented on src/plugins/console/public/lib/mappings/mappings.js:
```diff
@@ -41,11 +39,14 @@ export function expandAliases(indicesOrAliases) {
+    if (last !== value) {
+      result.push(value);
+    }
+    last = value;
```
nit: I'm not fully understand what happens here. I'm not a fan of retrieving and modifying any vars in parent scope of reduce function.  

If for some reason you need it, please use 3rd (index) and 4nd (array) arguments of reduce funciton

alexwizp commented on src/plugins/console/public/lib/mappings/mappings.js:
```diff
@@ -81,8 +82,8 @@ export function getFields(indices, types) {
-      $.each(typeDict, function (type, fields) {
-        if (!types || types.length === 0 || $.inArray(type, types) !== -1) {
+      Object.entries(typeDict).forEach(([type, fields]) => {
+        if (!types || types.length === 0 || types.findIndex((t) => t === type) !== -1) {
```
nit: can we use `.includes` instead of  `findIndex !== -1`?

alexwizp commented on src/plugins/console/public/application/containers/settings.tsx:
```diff
@@ -8,6 +8,7 @@
 
 import React from 'react';
 
+import { HttpSetup } from 'kibana/public';
```
nit: recommendation for future if you need to import only type please use the following format: 
`import type { HttpSetup } from 'kibana/public';`

alexwizp reviewed (commented):
Completed the first round of the review. Please review and correct comments

mibragimov commented:
@elasticmachine merge upstream

kibanamachine commented:
expected head sha didn’t match current head ref.

mibragimov commented:
@elasticmachine merge upstream

mibragimov commented:
@elasticmachine merge upstream

alexwizp commented on src/plugins/console/public/lib/es/es.ts:
```diff
@@ -24,44 +19,37 @@ export function getContentType(body: unknown) {
-    type: 'POST',
-    dataType: 'text', // disable automatic guessing
-  };
+export const initHttp = (http: HttpSetup) => {
```
can we avoid doing it? We are trying to avoid of setting services using that way. Please use more legal way for doing it

yuliacech commented on src/plugins/console/public/application/hooks/use_send_current_request_to_es/send_request_to_es.ts:
```diff
@@ -107,37 +115,48 @@ export function sendRequestToES(args: EsRequestArgs): Promise<ESRequestResult[]>
+            value,
+            contentType,
+            timeMs: Date.now() - startTime,
+            statusCode: error?.response?.status ?? 0,
```
nit: I think we should use one of the defined error status codes here instead of `0` as a default, maybe `500` would make more sense?

yuliacech commented on src/plugins/console/public/application/hooks/use_send_current_request_to_es/send_request_to_es.ts:
```diff
@@ -65,23 +67,29 @@ export function sendRequestToES(args: EsRequestArgs): Promise<ESRequestResult[]>
-          }
 
-          const xhr = dataOrjqXHR.promise ? dataOrjqXHR : jqXhrORerrorThrown;
+      try {
```
I'm most concerned about this file being changed without any test coverage. There is some logic about processing successful and failed requests that we should be testing I think.

mibragimov commented:
Thanks you for the review @yuliacech, I've addressed the comments you left with [06a1a43](https://github.com/elastic/kibana/pull/127867/commits/06a1a43a1e44b510cb46ba76cfa8323d8c706291) Let me know what you think.

mibragimov commented:
@elasticmachine merge upstream

sabarasaba reviewed (commented):
Thanks for having a go at this @mibragimov! Cannot explain how happy I am that jquery is so close to being gone, finally! Left a few comments, tested locally.

mibragimov commented:
Thanks for the review @sabarasaba! I've updated the PR with the suggested [changes](https://github.com/elastic/kibana/pull/127867/commits/a6e826571f6ec3edf63d640ea8b2ac6ef5cf4eec)

mibragimov commented:
@elasticmachine merge upstream

mibragimov commented:
@elasticmachine merge upstream

mibragimov commented:
@elasticmachine merge upstream

sabarasaba commented on src/plugins/console/common/constants/api.ts:
```diff
@@ -0,0 +1,9 @@
+ * Side Public License, v 1.
+ */
+
+export const API_BASE_PATH = '../api/console/proxy';
```
Base path should only be the base path without any extra routes on it, so that later on in our code we can refer these as `${API_BASE_PATH}/proxy` for example. This is a common pattern that we use in pretty much all of the other apps.

So I think we would want this to be:

```javascript
export const API_BASE_PATH = '/api/console';
```

sabarasaba reviewed (approved):
Thanks for addressing the feedback and adding tests for `send_request_to_es.test` @mibragimov! I left one small thing that I think we should correct before merging, but other than that the rest lgtm 🥳

sabarasaba commented:
Another thing, `v8.2.0` is past feature freeze so we should only target `v8.3.0`. Also since this is a PR that is an internal enhancement that is not user facing, I believe we would want to use the `release note: skip` tag instead of the one we have now.
